ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1235,28 +1235,41 @@ def test_sosreport_includes_healthcheck(self, create_logfile):
         )
         assert msg in cmd.stdout_text
 
-    def test_ipahealthcheck_verify_perms_for_source_files(self,
-                                                          modify_permissions):
+    def modify_perms_run_healthcheck(self, filename, modify_permissions):
         """
-        This tests checks if files in /var/log are checked with ipa.files
-        source.
-        The test modifies permissions of ipainstall log file and checks the
-        response from healthcheck.
-
-        https://pagure.io/freeipa/issue/8949
+        Modify the ipa logfile permissions and run
+        healthcheck command to check the status.
         """
-        modify_permissions(self.master, path=paths.IPASERVER_INSTALL_LOG,
-                           mode="0644")
+        modify_permissions(self.master, path=filename,
+                           mode="644")
         returncode, data = run_healthcheck(
             self.master, "ipahealthcheck.ipa.files", failures_only=True)
-
         assert returncode == 1
         assert len(data) == 1
         assert data[0]["result"] == "WARNING"
-        assert data[0]["kw"]["path"] == paths.IPASERVER_INSTALL_LOG
+        assert data[0]["kw"]["path"] == filename
         assert data[0]["kw"]["type"] == "mode"
         assert data[0]["kw"]["expected"] == "0600"
 
+    def test_ipahealthcheck_verify_perms_for_source_files(self):
+        """
+        This tests checks if files in /var/log are checked with ipa.files
+        source.
+        The test modifies permissions of ipainstall log file and checks the
+        response from healthcheck.
+        https://pagure.io/freeipa/issue/8949
+        """
+        self.modify_perms_run_healthcheck(paths.IPASERVER_INSTALL_LOG, "644")
+
+    def test_ipahealthcheck_verify_perms_upgrade_log_file(self):
+        """
+        This testcase creates /var/log/ipaupgrade.log file.
+        Once the file is generated the permissions are modified
+        to check that correct status message is displayed
+        by healthcheck tool
+        """
+        self.master.run_command(['touch', paths.IPAUPGRADE_LOG])
+        self.modify_perms_run_healthcheck(paths.IPAUPGRADE_LOG, "644")
```

Comment:
The second argument of modify_perms_run_healthcheck is a function, not a string. You need to call self.modify_perms_run_healthcheck(paths.IPAUPGRADE_LOG, modify_permissions).